Objects/frameobject.c

```diff
@@ -686,7 +686,6 @@ frame_clear(PyFrameObject *f, PyObject *Py_UNUSED(ignored))
     }
     if (f->f_frame->generator) {
         _PyGen_Finalize(f->f_frame->generator);
-        assert(f->f_frame->generator == NULL);
```

Comment:
The check doesn't make sense, `frame->generator` consistency is checked by `InterpreterFrame` (Python/frame.c), not by `PyFrameObject` (Objects/frameobject.c).

@markshannon I very much appreciate it if you find time for the review.

Reply:
I agree. This assert would be appropriate in the `dealloc` function, but not in `clear`.

Reply:
Brilliant idea. Added assertion to `frame_dealloc()`.